doc/dev/libraries.md

```diff
@@ -0,0 +1,84 @@
+- `libelektra-core`:
+  The core minimal API of Elektra.
+  Defines what `ElektraKey` and `ElektraKeyset` are, and contains the minimal API for manipulating them.
+
+  **Prefixes:** `elektraKey*` and `elektraKeyset*`
+
+- `libelektra-kdb`:
+  The main API for interfacing with the KDB.
+
+  **Prefix:** `elektraKdb*`
+
+- `libelektra-lowlevel-c`:
+  Additional C APIs that are useful when working with `ElektraKey` and `ElektraKeyset`, but not are not minimal API.
+  Specifically targets C and not intended for use via bindings.
+
+  **Prefix:** `elektra*`
+
+- `libelektra-highlevel-c`:
+  The C high-level API for reading/writing configuration with Elektra.
+  Intended for use by applications.
+  Partially intended for use with code-generation.
+  Specifically targets C and not intended for use via bindings.
+
+  **Prefix:** `elektraHlc*` (better suggestions welcome)
+  (Note: Even though there isn't any other `highlevel` implementation, I added the `-c` suffix to show this lib is only for C)
+
+- `libelektra-opts`:
+  The API for parsing command-line arguments according to Elektra's spec.
+  **Prefix:** `elektraOpts*`
+- `libelektra-notification`:
+  The API for setting up callbacks and automatically updated variables linked to changes in the KDB.
+
+  **Prefix:** `elektraNotification*`
+
+- `libelektra-io`:
+  Elektra's API for asynchronous operations.
+
+  **Prefix:** `elektraIo*`
+
+- `libelektra-merge`:
+  The API for merging and/or detecting conflicts between two keysets.
+
+  **Prefix:** `elektraMerge`
+
+- `libelektra-pluginbase`:
+  The API _used by_ plugins.
+
+  **Prefix:** `elektraPlugin*`
+  (Note: previously `libelektra-plugin`)
```

Comment:
I actually think the name `libelektra-pluginbase` makes more sense, because it forms the base for all plugins. It contains functions like `elektraPluginExport` and `elektraPluginGetData`.

The `libelektra-plugin` meanwhile is for interacting with plugins (not just from tooling) so the name makes sense too. It will contain things like `elektraPluginCall2Arg` to call a function from a plugin, possibly from another one.

Reply:
:+1: Ok, this makes sense. Please also add it to [doc/dev/libraries.md](https://github.com/ElektraInitiative/libelektra/pull/4245/files/6141c6adc9a7a166451f3292e3025db0c4ef3f48..44e6293f42f57d8dba0b397e01b0b61f5be0ccb1#diff-db061ad3ca3bb4e7f8806c2a6339502e77689bd7ecc8aa2bd79ba94345fb0948)